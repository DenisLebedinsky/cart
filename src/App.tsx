import * as React from "react";
import { connect } from "react-redux";

import CartContainer from "./containers/CartContainer";
import { getCart } from "./redux/reducers/cart/actions";

import css from "./_App.module.scss";

import { AppProps } from "./App.type";

export class App extends React.Component<AppProps> {
  public componentDidMount() {
    this.props.getCart();
  }

  public render() {
    return (
      <div className={css.app}>
        <CartContainer />
      </div>
    );
  }
}

const mapDispatchToProps = {
  getCart
};

export default connect(
  null,
  mapDispatchToProps
)(App);
