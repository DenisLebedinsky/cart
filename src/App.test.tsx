import * as React from "react";
import { shallow } from "enzyme";

import { App } from "./App";

describe("App", () => {
  it("should be defined", () => {
    const mock = jest.fn();
    const wrapper = shallow(<App getCart={mock} />).instance();

    expect(wrapper).toBeDefined();
  });

  it("should match the snapthot", () => {
    const mock = jest.fn();
    const wrapper = shallow(<App getCart={mock} />);

    expect(wrapper).toMatchSnapshot();
  });
});
