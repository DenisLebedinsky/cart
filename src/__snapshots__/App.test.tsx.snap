// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapthot 1`] = `
<div
  className="app"
>
  <ConnectFunction />
</div>
`;
